examples/teacache.py

```diff
@@ -0,0 +1,400 @@
+from typing import Any, Callable, Optional, Union
+
+import numpy as np
+import torch
+from diffusers.models.modeling_outputs import Transformer2DModelOutput
+from diffusers.models.transformers.transformer_flux import FluxTransformer2DModel
+from diffusers.utils import logging
+from diffusers.utils.constants import USE_PEFT_BACKEND
+from diffusers.utils.import_utils import is_torch_version
+from diffusers.utils.peft_utils import scale_lora_layers, unscale_lora_layers
+
+from nunchaku import NunchakuFluxTransformer2dModel  # type: ignore
+
+logger = logging.get_logger(__name__)  # pylint: disable=invalid-name
+
+
+def make_teacache_forward(
```

Comment:
Lastly, can we put in `nunchaku/models/transformers/transformer_flux` (btw I don't think it's best place to put this) or somewhere you believe more appropriate?

Reply:
I think this should stay as an example. I don't think it makes sense for you to bring TeaCache into your codebase. What does make sense IMO is to expose the layers so that you are API compatible with diffusers, and then any technique that uses these layers will be automatically compatible with Nunchaku